services/provider/server/server.go

```diff
@@ -201,3 +216,56 @@ func newClient() (client.Client, error) {
 
 	return client, nil
 }
+
+func (s *OCSProviderServer) getPublicKey(ctx context.Context, secretName string) (string, error) {
+	pubKeySecret := &corev1.Secret{}
+	err := s.client.Get(ctx, types.NamespacedName{Name: secretName, Namespace: s.namespace}, pubKeySecret)
+	if err != nil {
+		return "", fmt.Errorf("failed to get public key secret %q", secretName)
+	}
+
+	publicKey := string(pubKeySecret.Data["key"])
+	if publicKey == "" {
+		return "", fmt.Errorf("public key is not found inside the secret %q", secretName)
+	}
+
+	return publicKey, nil
+}
+
+func validateTicket(ticket, rawPubKey string) error {
+	block, _ := pem.Decode([]byte(rawPubKey))
+	if block == nil {
+		return fmt.Errorf("invalid PEM block")
+	}
+
+	key, err := x509.ParsePKIXPublicKey(block.Bytes)
+	if err != nil {
+		return fmt.Errorf("failed to parse public key. %v", err)
+	}
+
+	pubKey := key.(*rsa.PublicKey)
+
+	ticketArr := strings.Split(string(ticket), ".")
+	if len(ticketArr) != 0 {
```

Comment:
```suggestion
	if len(ticketArr) == 0 {
```

Reply:
I think a better test will be 
```suggestion
	if len(ticketArr) != 2 {
```
As we do assume (later in the function), exactly 2 parts.

Reply:
done